core/src/main/java/cz/o2/proxima/storage/StorageFilter.java

```diff
@@ -28,7 +28,7 @@
 public interface StorageFilter extends Serializable {
 
   /** Filter consisting of several filters with applied function. */
-  public abstract static class CompoundFilter implements StorageFilter {
```

Comment:
Don't do that. I think that this is constructed by reflection and will cause exception.